### ISSUE TITLE
Module not availble during start and restart

### ISSUE THREAD

**Issue by kismat1330** [tempesta-tech/tempesta#2026]
rmmod: ERROR: Module tempesta_fw is not currently loaded
Dec 26 12:03:14  tempesta.sh[6247]: rmmod: ERROR: Module tempesta_db is not currently loaded
Dec 26 12:03:14  tempesta.sh[6248]: rmmod: ERROR: Module tempesta_tls is not currently loaded
Dec 26 12:03:14 tempesta.sh[6249]: rmmod: ERROR: Module tempesta_lib is not currently loaded

when i try sudo systemctl status tempesta-fw or start again then its not working. Can anyone provide solution for same.   i am using ubunut 22.04.

**Comment by kismat1330** [comment]
also i am getting this error whene i am installing using sudo ./tempesta_installer.sh --install
ERROR (dkms apport): kernel package linux-headers-5.10.35.tfw-4c9ba16 is not supported
Error! Bad return status for module build on kernel: 5.10.35.tfw-4c9ba16 (x86_64)
Consult /var/lib/dkms/tempesta-fw/0.7.0/build/make.log for more information.
dpkg: error processing package tempesta-fw-dkms (--install):
 installed tempesta-fw-dkms package post-installation script subprocess returned error exit status 10
Errors were encountered while processing:
 tempesta-fw-dkm

so provide solution for this also.

**Comment by krizhanovsky** [comment]
It seems Tempesta FW was installed using the packages. @RomanBelozerov @const-t do you have any thoughts on this?

**Comment by kismat1330** [comment]
yes sir, we wants to use tempesta in place of nginx but before we working on POC of tempesta but its not working wiht pacakages. so how can install this on my ubunut 22.04 or 20.04 ??.

**Comment by RomanBelozerov** [comment]
@kismat1330 Hello. Please try to install using [Manual installation of prebuilt packages](https://github.com/tempesta-tech/tempesta/wiki/Install-from-packages#manual-installation-of-prebuilt-packages). Now we have same problem with our Nexus repo and the `/tempesta_installer.sh` script is unstable.

**Comment by kismat1330** [comment]
@RomanBelozerov according to manual installation still i am getting same error now what i have to do. Can you provide proper installation step for 22.04 or 20.04.

**Comment by RomanBelozerov** [comment]
@kismat1330 I have a questions:
1. do you checked  [requirements](https://github.com/tempesta-tech/tempesta/wiki/Requirements#common-requirements)?
2. log from `/var/lib/dkms/tempesta-fw/0.7.0/build/make.log`
3. what virtual machine are you using?

**Comment by kismat1330** [comment]
@RomanBelozerov
 1.   yes i have fullfill requirment 
2.  i am using ubunutu 22.04.2 (gcc 11.4.0) and i not using any VM ubunutu is directly installed in my pc with OS 64 bit, RAM 8 GB , INTEL aTOM CPU C3558@2.20GHz and disk capacity 168.1gb

but i am getting same error i follow all step.

error 
ERROR (dkms apport): kernel package linux-headers-5.10.35.tfw-4c9ba16 is not supported
Error! Bad return status for module build on kernel: 5.10.35.tfw-4c9ba16 (x86_64)
Consult /var/lib/dkms/tempesta-fw/0.7.0/build/make.log for more information.
dpkg: error processing package tempesta-fw-dkms (--install):
installed tempesta-fw-dkms package post-installation script subprocess returned error exit status 10
Errors were encountered while processing:
tempesta-fw-dkm

**Comment by kismat1330** [comment]
~/Downloads$ sudo systemctl status tempesta-fw
× tempesta-fw.service - High performance HTTP proxy server
     Loaded: loaded (/lib/systemd/system/tempesta-fw.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Fri 2023-12-29 11:45:47 IST; 2s ago
    Process: 93006 ExecStart=/lib/tempesta/scripts/tempesta.sh --start (code=exited, status=1/FAILURE)
    Process: 93085 ExecStopPost=/lib/tempesta/scripts/tempesta.sh --stop (code=exited, status=0/SUCCESS)
   Main PID: 93006 (code=exited, status=1/FAILURE)
        CPU: 269ms

Dec 29 11:45:47 sdwanlab tempesta.sh[93085]: ...unload Tempesta modules
Dec 29 11:45:47 sdwanlab tempesta.sh[93085]: Un-loading Tempesta kernel modules...
Dec 29 11:45:47 sdwanlab tempesta.sh[93100]: rmmod: ERROR: Module tempesta_fw is not currently loaded
Dec 29 11:45:47 sdwanlab tempesta.sh[93101]: rmmod: ERROR: Module tempesta_db is not currently loaded
Dec 29 11:45:47 sdwanlab tempesta.sh[93102]: rmmod: ERROR: Module tempesta_tls is not currently loaded
Dec 29 11:45:47 sdwanlab tempesta.sh[93104]: rmmod: ERROR: Module tempesta_lib is not currently loaded
Dec 29 11:45:47 sdwanlab tempesta.sh[93085]: ...revert irqbalance config
Dec 29 11:45:47 sdwanlab tempesta.sh[93085]: done
Dec 29 11:45:47 sdwanlab systemd[1]: tempesta-fw.service: Failed with result 'exit-code'.
Dec 29 11:45:47 sdwanlab systemd[1]: Failed to start High performance HTTP proxy server.

**Comment by RomanBelozerov** [comment]
@kismat1330 I suggest you try installing TempestaFW from sources:
1. boot into the system using kernel 5.10.35.tfw-4c9ba16
2. install build dependences
```
apt-get install build-essential libboost-dev libboost-program-options-dev
```
3. download TempestaFW
```
git clone -b release-0.7 https://github.com/tempesta-tech/tempesta
cd tempesta
```
4.  run build as root
```
make clean && make
```
5. run TempestaFW as root
```
./scripts/tempesta.sh --start
```

Probably you should try installing on a clean ubuntu 22.

**Comment by kismat1330** [comment]
sdwanlab@sdwanlab:~/tempesta$ sudo ./scripts/tempesta.sh --start
Starting Tempesta...
...enable RPS on enp7s0
...enable RPS on enp8s0
...enable RPS on enp9s0
...enable RPS on enp10s0
...enable RPS on enp13s0
...enable RPS on enp14s0
...enable RPS on enp15s0
...enable RPS on enp16s0
...enable RPS on eno1
...enable RPS on eno2
...load Tempesta modules
Loading Tempesta kernel modules...
Loading module tempesta_lib 
insmod: ERROR: could not load module /home/sdwanlab/tempesta/lib/tempesta_lib.ko: No such file or directory
ERROR: cannot load tempesta library module

**Comment by EvgeniiMekhanik** [comment]
@kismat1330 Was the build successful? If yes, where is the file "tempesta_lib.ko" located in your system?

**Comment by krizhanovsky** [comment]
Hi @kismat1330 ,

I'm wondering if you tried to install Tempesta FW using the script https://github.com/tempesta-tech/tempesta/wiki/Install-from-packages#using-installer-script ?

@ykargin created #2028 for both this and #2027 issues - probably we should leave only one way for the binary installation, the script way, as more straightforward and debuggable.

**Comment by kismat1330** [comment]
@krizhanovsky, sir i tryed all the way using binary, source and manual but not getting success. 
i am using ubuntu 22.04.

**Comment by ykargin** [comment]
Hi @kismat1330, I've updated the installation script, but as far as I understand from #2035, the installation issue has been resolved, right?

**Comment by ykargin** [comment]
Hi @kismat1330,

It looks like the issue has been resolved, so I am closing this bug. However, if you have any additional information or if the issue persists, please feel free to reopen this issue. Thank you for your assistance in improving our project.
